.github/workflows/build.yml

```diff
@@ -37,81 +39,51 @@ jobs:
         with:
           toolchain: stable
           override: true
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v3
         name: Setup Node
         with:
-          node-version: "*"
-      - name: Install Yarn
-        run: corepack enable
+          node-version: 16
+          cache: yarn
       - name: Install npm dependencies
-        run: yarn install
-      - name: Install tauri dependencies
-        run: sudo apt update && sudo apt install -y build-essential libpython3-dev libdbus-1-dev libgtk-3-dev libsoup-gnome2.4-dev libwebkit2gtk-4.0-dev
+        run: yarn install --immutable --check-cache
+      - if: matrix.os == 'ubuntu-20.04'
+        name: Install Build Packages
+        uses: awalsh128/cache-apt-pkgs-action@latest
+        with:
+          packages: libpython3-dev libdbus-1-dev libgtk-3-dev libsoup-gnome2.4-dev libwebkit2gtk-4.0-dev
+          version: 1.0
+      - uses: Swatinem/rust-cache@v2
       - name: Build Tauri
         run: yarn run build:tauri
-      - uses: actions/upload-artifact@v3
+      - if: matrix.os == 'macos-latest'
+        name: Compress App
+        run: tar -cvzf qcvis.app.tar.gz ./src-tauri/target/release/bundle/macos/qcvis.app
+      - if: matrix.os == 'ubuntu-20.04'
+        uses: actions/upload-artifact@v3
         name: Upload Artifacts (appimage)
         with:
           name: tauri-linux
           path: src-tauri/target/release/bundle/appimage/*.AppImage
-      - uses: actions/upload-artifact@v3
+      - if: matrix.os == 'ubuntu-20.04'
+        uses: actions/upload-artifact@v3
```

Comment:
Can't the uploads be bundled into a single one with multiple globs and a parametric name?

Reply:
I don't have an issue with using the OS name in the artifact name. For macOS, the directory compression can be done conditionally. I'm not against conditionals in general, I just want to minimize them here if possible